mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -434,9 +434,7 @@ arith::MulSIExtendedOp::fold(FoldAdaptor adaptor,
     // Invoke the constant fold helper again to calculate the 'high' result.
     Attribute highAttr = constFoldBinaryOp<IntegerAttr>(
         adaptor.getOperands(), [](const APInt &a, const APInt &b) {
```

Comment:
Maybe this can just be `...(adaptor.getOperands(), llvm::APIntOps::mulhs);`?

Reply:
> Maybe this can just be `...(adaptor.getOperands(), llvm::APIntOps::mulhs);`?

Not sure, I think it is clear as it is.

Reply:
I'd recommend any further refactor changes like that as followups for somebody (not necessarily @Atousa) - to ensure the changes in this PR are just the move to the APIntOps::mulh* helpers.